### PR TITLE
Scripting promote video-stream-information to an applet

### PR DESCRIPTION
**Pull request by rmackay9**
This enhances the set_VIDEO_STREAM_INFORMATION.lua to make it usable without any code changes by at least some users.  The script is also promoted to an applet

The purpose of this script is to give the AP_Camera library the information it needs so that it can send [VIDEO_STREAM_INFORMATION](https://mavlink.io/en/messages/common.html#VIDEO_STREAM_INFORMATION) mavlink messages to the ground station so that it can display the live video stream to the user

15 new parameters have been added prefixed with VID1_ that allow the user to set nearly all fields of the VIDEO_STREAM_INFORMATION but for most users all they will only need to set these six:

- VID1_CAMMODEL: Video stream camera model (1:Siyi A8, 2:Siyi ZR10, 3:Siyi ZR30, 4:Siyi ZT30 Zoom, 5:Siyi ZT30 Wide, 6:Siyi ZT30 IR, 7:Siyi ZT6 RGB, 8:Siyi ZT6 IR)
- VID1_IPADDR0: Video stream IP Address first octet
- VID1_IPADDR1: Video stream IP Address second octet
- VID1_IPADDR2: Video stream IP Address third octet
- VID1_IPADDR3: Video stream IP Address fourth octet
- VID1_IPPORT: Video Stream IP Address Port

This has been successfully tested with the Siyi A8, ZR10 and ZT6 cameras and [Mission Planner's new gimbal interface](https://github.com/ArduPilot/MissionPlanner/pull/3386)

Enhancements coming in future PRs:

- default the IP address and Port to the camera defaults
- add support for ViewPro, Topotek and any other Ethernet enabled camera gimbal

### CODE REVIEW

File: libraries/AP_Scripting/applets/video-stream-information.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-- VID1_IPADDR2: Video stream IP Address third octet
+-- VID1_IPADDR3: Video stream IP Address fourth octet
+-- VID1_IPPORT: Video Stream IP Address Port
+
+# How To Use
+
+1. setup the camera gimbal as described on the ArduPilot wiki including ethernet setup
+2. check the IP address of the camera gimbal
+3. copy this script to the vehicle autopilot's "scripts" directory
+4. set the VID1_CAMMODEL parameter to the camera gimbal model
```
Comment (Hwurzburg) [inline libraries/AP_Scripting/applets/video-stream-information.md:29]:
```suggestion
4. setup scripting per https://ardupilot.org/plane/docs/common-lua-scripts.html and reboot . Set the VID1_CAMMODEL parameter to the camera gimbal model and adjust the new VID1 params as required.
```

**Review by Hwurzburg — DISMISSED** [review]

File: libraries/AP_Scripting/applets/video-stream-information.lua [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    local uri_prefix = ''
+    if VID1_TYPE:get() == VID_TYPE_ENUM.RTSP then
+        uri_prefix = 'rtsp://'
+    end
+    local uri_ip = math.floor(VID1_IPADDR0:get()) .. '.' ..
+                   math.floor(VID1_IPADDR1:get())  .. '.' ..
+                   math.floor(VID1_IPADDR2:get())  .. '.' ..
+                   math.floor(VID1_IPADDR3:get())  .. ':' ..
+                   math.floor(VID1_IPPORT:get())
+    local uri = uri_prefix .. uri_ip .. uri_suffix
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Scripting/applets/video-stream-information.lua:213]: cool, fixed, thanks!

File: libraries/AP_Scripting/applets/video-stream-information.md (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Scripting/applets/video-stream-information.md:29]: great, adjusted the how-to-use instructions, thanks!

File: libraries/AP_Scripting/applets/video-stream-information.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Video Stream Information
+
+This script updates the ArduPilot Camera library with the values required to populate the VIDEO_STREAM_INFORMATION
+mavlink message allow the GCS to find and display live video to the user
+
+# Parameters
+
+-- VID1_CAMMODEL: Video stream camera model (1:Siyi A8, 2:Siyi ZR10, 3:Siyi ZR30, 4:Siyi ZT30 Zoom, 5:Siyi ZT30 Wide, 6:Siyi ZT30 IR, 7:Siyi ZT6 RGB, 8:Siyi ZT6 IR)
```
Comment (tridge) [inline libraries/AP_Scripting/applets/video-stream-information.md:8]: doesn't look right in view file

Comment (rmackay9) [reply libraries/AP_Scripting/applets/video-stream-information.md]: fixed, thanks!

**Comment by rmackay9** [comment]
Since yesterday's review, I've also added support for Herelink, Topotek and ViewPro.  During testing Topotek and ViewPro worked as expected but Herelink is giving some troubles.  Still, unless there are objections I will merge this once it passes CI.

**Comment by rmackay9** [comment]
FYI @nexton-winjeel

**Pull request merged**